File: Java/RecyclerView16RecyclerViewPinterestCells/app/src/main/java/dev/ogabek/recyclerview16_recyclerviewpinterestcells/activity/MainActivity.java
package dev.ogabek.recyclerview16_recyclerviewpinterestcells.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import android.content.Context;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

import dev.ogabek.recyclerview16_recyclerviewpinterestcells.R;
import dev.ogabek.recyclerview16_recyclerviewpinterestcells.adapter.MainRecyclerViewAdapter;
import dev.ogabek.recyclerview16_recyclerviewpinterestcells.helper.SpacesItemDecoration;
import dev.ogabek.recyclerview16_recyclerviewpinterestcells.model.Member;

public class MainActivity extends AppCompatActivity {

    Context context;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        refreshAdapter(prepareMemberList());

    }

    private List<Member> prepareMemberList() {
        List<Member> members = new ArrayList<>();
        for (int i = 0; i < 50; i++) {
            members.add(new Member());
        }
        return members;
    }

    private void initViews() {
        context = this;
        recyclerView = findViewById(R.id.rv_mainRecyclerView);
        recyclerView.setLayoutManager(new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL));

        SpacesItemDecoration decoration = new SpacesItemDecoration(10);
        recyclerView.addItemDecoration(decoration);

    }

    private void refreshAdapter(List<Member> members) {
        MainRecyclerViewAdapter adapter = new MainRecyclerViewAdapter(context, members);
        recyclerView.setAdapter(adapter);
    }

}

File: Kotlin/RecyclerView16RecyclerViewPinterestCall/app/src/main/java/dev/ogabek/recyclerview_16_recyclerviewpinterestcall/activity/MainActivity.kt
package dev.ogabek.recyclerview_16_recyclerviewpinterestcall.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.StaggeredGridLayoutManager
import dev.ogabek.recyclerview_16_recyclerviewpinterestcall.R
import dev.ogabek.recyclerview_16_recyclerviewpinterestcall.adapter.MainAdapter
import dev.ogabek.recyclerview_16_recyclerviewpinterestcall.helper.SpacesItemDecoration
import dev.ogabek.recyclerview_16_recyclerviewpinterestcall.model.Member

class MainActivity : AppCompatActivity() {

    private lateinit var context: Context
    private lateinit var recyclerView: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        refreshAdapter(prepareMemberList())

    }

    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.rv_main_recycler_view)
        recyclerView.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)

        val spacesItemDecoration = SpacesItemDecoration(10)
        recyclerView.addItemDecoration(spacesItemDecoration)

    }

    private fun refreshAdapter(members: List<Member>) {
        val adapter = MainAdapter(context, members as ArrayList<Member>)
        recyclerView.adapter = adapter
    }

    private fun prepareMemberList(): List<Member> {
        val members = ArrayList<Member>()
        for (i in 0..100) members.add(Member())
        return members
    }

}